chain/client/src/gc_actor.rs

```diff
@@ -0,0 +1,102 @@
+use crate::metrics;
+use actix::{Actor, Addr, Arbiter, ArbiterHandle, AsyncContext, Context};
+use near_chain::{types::RuntimeAdapter, ChainStore, ChainStoreAccess};
+use near_chain_configs::{ClientConfig, GCConfig};
+use near_epoch_manager::EpochManagerAdapter;
+use near_primitives::types::BlockHeight;
+use near_store::{metadata::DbKind, Store};
+use std::sync::Arc;
+use tracing::warn;
+
+pub struct GCActor {
+    store: ChainStore,
+    runtime_adapter: Arc<dyn RuntimeAdapter>,
+    epoch_manager: Arc<dyn EpochManagerAdapter>,
+    gc_config: GCConfig,
+    is_archive: bool,
+}
+
+impl GCActor {
+    pub fn new(
+        store: Store,
+        genesis_height: BlockHeight,
+        runtime_adapter: Arc<dyn RuntimeAdapter>,
+        epoch_manager: Arc<dyn EpochManagerAdapter>,
+        gc_config: GCConfig,
+        is_archive: bool,
+    ) -> Self {
+        GCActor {
+            store: ChainStore::new(store, genesis_height, true),
+            runtime_adapter,
+            gc_config,
+            epoch_manager,
+            is_archive,
+        }
+    }
+
+    fn gc(&mut self, ctx: &mut Context<GCActor>) {
+        let timer = metrics::GC_TIME.start_timer();
+        if !self.is_archive {
+            let _span = tracing::debug_span!(target: "client", "gc");
+            if let Err(e) = self.store.clear_data(
+                &self.gc_config,
+                self.runtime_adapter.clone(),
+                self.epoch_manager.clone(),
+            ) {
+                warn!(target: "client", "Error in gc: {}", e);
+            }
+        } else {
+            let _span = tracing::debug_span!(target: "client", "archive_gc");
+            let kind = self.store.store().get_db_kind();
+            match kind {
+                Ok(Some(DbKind::Hot)) => {
+                    if let Err(e) = self.store.clear_data(
+                        &self.gc_config,
+                        self.runtime_adapter.clone(),
+                        self.epoch_manager.clone(),
+                    ) {
+                        warn!(target: "client", "Error in gc: {}", e);
+                    }
+                }
+                Err(e) => {
+                    warn!(target: "client", "Error in gc: {}", e);
+                }
+                _ => {}
```

Comment:
I believe for other DbKind types you should call the `clear_archive_data` method.

Reply:
cc @posvyatokum who's an expert and a recent editor of DbKinds :)